src/txn/txn_rollback_to_stable.c

```diff
@@ -1213,32 +1207,64 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
         WT_ERR(__wt_config_getones(session, config, "checkpoint", &cval));
         __wt_config_subinit(session, &ckptconf, &cval);
         for (; __wt_config_next(&ckptconf, &key, &cval) == 0;) {
-            ret = __wt_config_subgets(session, &cval, "newest_start_durable_ts", &durableval);
+            ret = __wt_config_subgets(session, &cval, "newest_start_durable_ts", &value);
             if (ret == 0) {
                 newest_start_durable_ts =
-                  WT_MAX(newest_start_durable_ts, (wt_timestamp_t)durableval.val);
+                  WT_MAX(newest_start_durable_ts, (wt_timestamp_t)value.val);
                 durable_ts_found = true;
             }
             WT_ERR_NOTFOUND_OK(ret, false);
-            ret = __wt_config_subgets(session, &cval, "newest_stop_durable_ts", &durableval);
+            ret = __wt_config_subgets(session, &cval, "newest_stop_durable_ts", &value);
             if (ret == 0) {
-                newest_stop_durable_ts =
-                  WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)durableval.val);
+                newest_stop_durable_ts = WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)value.val);
                 durable_ts_found = true;
             }
             WT_ERR_NOTFOUND_OK(ret, false);
-            ret = __wt_config_subgets(session, &cval, "prepare", &durableval);
+            ret = __wt_config_subgets(session, &cval, "prepare", &value);
             if (ret == 0) {
-                if (durableval.val)
+                if (value.val)
                     prepared_updates = true;
             }
             WT_ERR_NOTFOUND_OK(ret, false);
+            ret = __wt_config_subgets(session, &cval, "addr", &value);
+            if (ret == 0)
+                addr_size = value.len;
+            WT_ERR_NOTFOUND_OK(ret, false);
         }
         max_durable_ts = WT_MAX(newest_start_durable_ts, newest_stop_durable_ts);
+
+        /*
+         * The rollback to stable will skip the tables during recovery and shutdown in the following
+         * conditions.
+         * 1. Empty table
+         * 2. Table has timestamped updates without a stable timestamp.
+         */
+        if ((F_ISSET(S2C(session), WT_CONN_RECOVERING) ||
```

Comment:
Checking for the empty file can be done in every RTS scenario. So adding a check for recovery and shutdown is not required. But I checked the test failures before this change is due to in-memory scenario. Change the condition to an empty file only in non-in-memory scenarios. Because the file size on the disk is always empty for in-memory scenarios.

Reply:
Thanks @kommiharibabu . Updated the code to skip processing RTS on empty tables and not the in-memory scenario.